hazelcast/src/main/java/com/hazelcast/sql/impl/expression/string/StringExpressionUtils.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.string;
+
+import com.hazelcast.sql.impl.QueryException;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.row.Row;
+
+import java.util.Locale;
+
+/**
+ * Utility methods for string functions.
+ */
+public final class StringExpressionUtils {
+    private StringExpressionUtils() {
+        // No-op.
+    }
+
+    public static String concat(String first, String second) {
+        return (first != null ? first : "") + (second != null ? second : "");
```

Comment:
if at least one operand is null, it should return null

Reply:
There is no such requirement in the standard. It is up to the vendor to decide how to treat `NULL` values here. E.g. PG and MySQL return `NULL` in this case, while Oracle and SQL Server treats `NULL` as an empty string. 
Why do you think that returning `NULL` is better than treating it as an empty string?

Reply:
From the standard:

`<concatenation> ::= <character value expression> <concatenation operator> <character factor>`
...
`Let S1 and S2 be the result of the <character value expression> and <character factor>, respectively.`
...
`If either S1 or S2 is the null value, then the result of the <concatenation> is the null value.`

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/17379/commits/2dab533ac3628b139b36fa06814602a8c4eba960